operator/internal/controller/redpanda/managed_decommission_controller.go

```diff
@@ -247,106 +236,100 @@ func decommissionStatus(ctx context.Context, l logr.Logger, c k8sclient.Client,
 	}
 }
 
-var ErrZeroReplicas = errors.New("redpanda replicas is zero")
-
-func getRedpandaReplicas(ctx context.Context, c k8sclient.Client, rp *v1alpha2.Redpanda) (int, error) {
-	requestedReplicas := ptr.Deref(rp.Spec.ClusterSpec.Statefulset.Replicas, 0)
-
-	if requestedReplicas == 0 {
-		resourcesName := rp.Name
-		if override := ptr.Deref(rp.Spec.ClusterSpec.FullnameOverride, ""); override != "" {
-			resourcesName = override
-		}
-
-		var sts appsv1.StatefulSet
-		err := c.Get(ctx, types.NamespacedName{
-			Namespace: rp.Namespace,
-			Name:      resourcesName,
-		}, &sts)
-		if err != nil {
-			return 0, fmt.Errorf("getting statefulset : %w", err)
-		}
-
-		requestedReplicas = int(ptr.Deref(sts.Spec.Replicas, 0))
-	}
-
-	if requestedReplicas == 0 {
-		return 0, ErrZeroReplicas
-	}
-	return requestedReplicas, nil
-}
-
-func podEvict(ctx context.Context, l logr.Logger, c k8sclient.Client, rp *v1alpha2.Redpanda) error {
+func (r *ManagedDecommissionReconciler) podEvict(ctx context.Context, rp *v1alpha2.Redpanda) error {
 	if rp.Status.ManagedDecommissioningNode == nil {
 		return nil
 	}
 
 	decommissionNodeID := int(*rp.Status.ManagedDecommissioningNode)
-	log := l.WithName("podEvict").WithValues("decommission-node-id", decommissionNodeID)
+	log := ctrl.LoggerFrom(ctx).WithName("podEvict").WithValues("decommission-node-id", decommissionNodeID)
 
-	pod, err := getPodFromRedpandaNodeID(ctx, l, c, rp, decommissionNodeID)
+	pod, err := r.getPodFromRedpandaNodeID(ctx, rp, decommissionNodeID)
 	if err != nil {
 		return fmt.Errorf("getting pod from node-id (%d): %w", decommissionNodeID, err)
 	}
 
+	// NB: Deleting this Pod will take an unexpectedly long time as the
+	// pre-stop hook will "spin" due to not handling decommissioned brokers.
```

Comment:
I don't understand this comment. I'm under impression that podEvict is called after Redpanda is successfully decommissioned. Maybe the problem is that maintenance mode is not achievable if particular Redpanda is no longer part of the Cluster? 

https://github.com/redpanda-data/redpanda-operator/blob/a36efbc5191d7b287ba8c6ad83645522357b6132/operator/internal/controller/redpanda/managed_decommission_controller.go#L208-L221

Reply:
> Maybe the problem is that maintenance mode is not achievable if particular Redpanda is no longer part of the Cluster?

Exactly this! You can't set maintenance mode on a broker that's not part of a cluster. The admin API returns a 404.